sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpLoggingPolicy.java

```diff
@@ -11,68 +11,57 @@
 import io.clientcore.core.http.models.HttpResponse;
 import io.clientcore.core.http.models.Response;
 import io.clientcore.core.implementation.http.HttpRequestAccessHelper;
-import io.clientcore.core.implementation.http.HttpResponseAccessHelper;
 import io.clientcore.core.implementation.util.ImplUtils;
 import io.clientcore.core.implementation.util.LoggingKeys;
 import io.clientcore.core.util.ClientLogger;
 import io.clientcore.core.util.binarydata.BinaryData;
 
 import java.io.IOException;
 import java.net.URI;
-import java.time.Duration;
-import java.util.Collections;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
-import static io.clientcore.core.http.models.ContentType.APPLICATION_OCTET_STREAM;
-import static io.clientcore.core.http.models.HttpHeaderName.CLIENT_REQUEST_ID;
 import static io.clientcore.core.http.models.HttpHeaderName.TRACEPARENT;
 import static io.clientcore.core.implementation.util.ImplUtils.isNullOrEmpty;
 
 /**
  * The pipeline policy that handles logging of HTTP requests and responses.
  */
 public class HttpLoggingPolicy implements HttpPipelinePolicy {
+    private static final HttpLogOptions DEFAULT_HTTP_LOG_OPTIONS = new HttpLogOptions();
+    private static final Set<HttpHeaderName> ALWAYS_ALLOWED_HEADERS = Set.of(TRACEPARENT);
     private static final int MAX_BODY_LOG_SIZE = 1024 * 16;
     private static final String REDACTED_PLACEHOLDER = "REDACTED";
     private static final ClientLogger LOGGER = new ClientLogger(HttpLoggingPolicy.class);
     private final HttpLogOptions.HttpLogDetailLevel httpLogDetailLevel;
     private final Set<HttpHeaderName> allowedHeaderNames;
+
     private final Set<String> allowedQueryParameterNames;
-    private final HttpRequestLogger requestLogger;
-    private final HttpResponseLogger responseLogger;
 
-    private static final String REQUEST_LOG_MESSAGE = "HTTP request";
-    private static final String RESPONSE_LOG_MESSAGE = "HTTP response";
+    private static final String HTTP_REQUEST_EVENT_NAME = "http.request";
+    private static final String HTTP_RESPONSE_EVENT_NAME = "http.response";
+
+    // request log level is low (verbose) since almost all request details are also
+    // captured on the response log.
```

Comment:
Are we confident in this? What if we're trying to capture logs for requests that never get a response / timely reponse?

Reply:
We should not have requests that disappear - they should get an exception after the timeout which we'll log when it happens.

We don't capture request headers and request body on the response log (we can though).
But the point is that response information contains almost everything you need to common analysis - you can enable verbose logs if you need extra.